cmd/ib-sriov-cni/main.go

```diff
@@ -112,31 +113,26 @@ func cmdAdd(args *skel.CmdArgs) error {
 	if netConf.RdmaIso {
 		err = utils.EnsureRdmaSystemMode()
 		if err != nil {
-			return err
+			return nil, nil, err
 		}
 	}
 
 	netns, err := ns.GetNS(args.Netns)
 	if err != nil {
-		return fmt.Errorf("failed to open netns %q: %v", netns, err)
+		return nil, nil, fmt.Errorf("failed to open netns %q: %v", netns, err)
 	}
-	defer netns.Close()
 
 	err = config.LoadDeviceInfo(netConf)
 	if err != nil {
-		return fmt.Errorf("failed to get device specific information. %v", err)
-	}
-
-	sm := sriov.NewSriovManager()
-
-	// Lock CNI operation to serialize the operation
-	lock, err := lockCNIExecution()
-	if err != nil {
-		return err
+		netns.Close()
```

Comment:
if you move `config.LoadDeviceInfo()` above `ns.GetNS` you would not need to do this